docs/proposals/validation-tools/readme.md

```diff
@@ -0,0 +1,83 @@
+# Swagger Validation Tools VS Code Extension
+
+## Objectives and High-level plan
+
+We plan on building a one-stop shop vs-code extension that will help the swagger authors get live and prompt feedback on the quality and correctness of their swaggers. We plan on reusing and building on top of the work that was done for the Autorest Linter VS code extension and integration our tools Autorest to simplify the users experience.
+
+## Overall architecture
+
+The following diagram provides a high-level architecture of the proposed SVT (Swagger Validation tools) VS Code extension.
+
+> ![](./diagram.png)
+
+1. SVT VS Code extension&#39;s client: the client is the implementation of UI and experience of the swagger authoring, details will follow later in the document on the proposed user experience. The swagger tooling team will take ownership of this component.
+
+2. SVT VS Code server component: The Autorest team will be owning this component and the current plan is to:
+
+  - Load up Autorest in memory of the machine where VS Code resides.
+  - Investigate a way to include the server logic or process within Autorest (basically turn Autorest into the server) to improve performance.
+  - Define a clear interface for the different operations and calls that the client side will need to trigger different validation tools.
+
+3. Pluggable linter rules: the Autorest team will be working on a design and interface to allow anyone to contribute static linting rules in a more scalable (not directly authored into the code) way. The swagger volition tools team or anyone from the services team will be able to contribute to the rules as they see fit.
+
+4. Pluggable Validation tools: The swagger tooling team will mainly be responsible for building these tools except for the linter which will continue to be owned by the Autorest team, but that wouldn&#39;t stop contribution from anyone else. in the efforts of simplifying the user experience and potentially simplifying the process of integrating our tools with any IDE in the future (such as Visual Studio), all the validation tools will be hooked up to extension points on Autorest which is currently work in progress on Autorests side. Our tools can all be run through simple commands but we&#39;re committed to update them to adhere with any integration requirements from Auto-rest.
+

```

Comment:
For swagger authors, we may also want to provide some type of navigation facility, to view/peek #definitions, or refer to defined ones (with intellisense), autocompletion, etc, to improve the creating/editing experience.